src/passes/inlineAssemblyTransformer.ts

```diff
@@ -0,0 +1,176 @@
+import { ASTMapper } from '../ast/mapper';
+import { AST } from '../ast/ast';
+import {
+  createExpressionStatement,
+  createIdentifier,
+  createNumberLiteral,
+  createTuple,
+} from '../utils/nodeTemplates';
+
+import {
+  InlineAssembly,
+  YulNode,
+  Identifier,
+  ASTNode,
+  BinaryOperation,
+  Assignment,
+  Block,
+  Expression,
+  ExpressionStatement,
+  VariableDeclaration,
+  UncheckedBlock,
+  Literal,
+  Statement,
+  ASTContext,
+} from 'solc-typed-ast';
+import assert from 'assert';
+import { getErrorMessage, WillNotSupportError } from '../utils/errors';
+
+const binaryOps: { [name: string]: string } = { add: '+', mul: '*', div: '/', sub: '-' };
```

Comment:
We are missing operations like shr shl etc
https://docs.soliditylang.org/en/v0.8.17/yul.html#evm-dialect